### PR TITLE
Added ui tests and yaml file for automation

### PR DESCRIPTION
**Pull request by ReannaMcE**

### CODE REVIEW

File: test/ui/ViewJobRolesTests.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            await ViewJobRolesTestsPage.assertInstagram();
+            await ViewJobRolesTestsPage.loadPage();
+            await ViewJobRolesTestsPage.assertJobRolesTitle();
+        } catch (error) {
+            console.error('Test failed:', error);
+            throw error;
+        }
+    });
+
+    // it('Job status should always be open', async function () {
```
Comment (shaunganley) [inline test/ui/ViewJobRolesTests.ts]: remove if not needed

File: test/ui/ViewJobRolesTestsPage.ts [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    public static async assertInstagram(): Promise<void> {
+        await this.driver.wait(async () => {
+            const currentUrl = await this.driver.getCurrentUrl();
+            return currentUrl.includes('instagram.com');
+        }, 15000);
+    }
+
+    // delete this when the id is available and replace with the test below
+    public static async assertJobRolesTitle(): Promise<void> {
+        const title = By.xpath("//h2[normalize-space()='Available Job Roles']");
```
Comment (shaunganley) [inline test/ui/ViewJobRolesTestsPage.ts]: use ID when finding, much more reliable than xpaths

File: test/ui/ViewJobRolesTestsPage.ts [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+            await this.driver.wait(until.elementIsVisible(instagramLink), 10000);
+            await instagramLink.click();
+        } catch (error) {
+            console.error('Error clicking the Instagram link:', error);
+            throw error;
+        }
+    }
+
+
+    // public static async clickHome(): Promise<void> {
```
Comment (shaunganley) [inline test/ui/ViewJobRolesTestsPage.ts]: clear out if not needed

**Review by shaunganley — CHANGES_REQUESTED** [review]
nice work. clear out commented out code. I suggest moving things like the social media links to a separate class named Footer which all of your pages could then extend (asks the devs if you're not sure how to do this). Add an ID to find anything on the page that your want to assert on

**Review by BenjaminBrashaw — APPROVED** [review]

**Review by BenjaminBrashaw — APPROVED** [review]

File: test/ui/FooterTestPage.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    // Assertions --------------------------------------------------------------------------------
+    public static async assertAndGoBack(actualUrl: string, page: string): Promise<void> {
+        await this.driver.wait(async () => {
+            const currentUrl = await this.driver.getCurrentUrl();
+            return currentUrl.includes(actualUrl);
+        }, 15000);
+
+        await FooterTestPage.loadPage(page);
+    }
```
Comment (shaunganley) [inline test/ui/FooterTestPage.ts]: these methods don't seem specific to the Footer? they look like re-usable methods that can be in your basepage

File: test/ui/ViewJobRolesTests.ts [hunk 5]
```diff
@@ -1,100 +1,61 @@
...
 
 describe('View Job Roles Tests', function () {
-    this.timeout(50000);
+    this.timeout(70000);
+    const page:string = 'https://jptw3amsi2.eu-west-1.awsapprunner.com/jobRoles';
 
     before(async function () {
         await ViewJobRolesTestsPage.startDriver();
+        FooterTestPage.driver = ViewJobRolesTestsPage.getDriver(); // the footer test uses the same driver
+        HeaderTestPage.driver = ViewJobRolesTestsPage.getDriver(); // the header test uses the same driver
```
Comment (shaunganley) [inline test/ui/ViewJobRolesTests.ts]: header and footer shouldn't need their own driver. instead have the ViewJobRolesTestsPage extend the header and footer and access the methods through ViewJobRolesTestsPage

File: test/ui/FooterTestPage.ts (hunk 4 above)
Comment (shaunganley) [reply test/ui/FooterTestPage.ts]: you should have a method called clickFacebookLink, clickTwitterLink etc which call the click link in the basepage

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: .github/workflows/testui.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
+name: Run UI Tests
+
+on: 
+  pull_request:
+    branches:
+      - main
```
Comment (shaunganley) [inline .github/workflows/testui.yaml]: don't run on pull request as these take too long

File: test/ui/HeaderTestPage.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    // Click Buttons --------------------------------------------------------------------------------
+    public static async clickButton(id: string, buttonName: string): Promise<void> {
+        try {
+            const btn: WebElement = await this.getButton(id);
+            await this.driver.wait(until.elementIsVisible(btn), 10000);
+            await btn.click();
+        } catch (error) {
+            console.error('Error clicking the' + buttonName + 'button:', error);
+            throw error;
+        }
```
Comment (shaunganley) [inline test/ui/HeaderTestPage.ts]: these should be in basepage

File: test/ui/ViewJobRolesTests.ts [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+import { basepage } from './basepage';
+
+describe('View Job Roles Tests', function () {
+    this.timeout(70000);
+    const page:string = 'https://jptw3amsi2.eu-west-1.awsapprunner.com/jobRoles';
+
+    before(async function () {
+        await basepage.startDriver();
+        //FooterTestPage.driver = ViewJobRolesTestsPage.getDriver(); // the footer test uses the same driver
+        //HeaderTestPage.driver = ViewJobRolesTestsPage.getDriver(); // the header test uses the same driver
```
Comment (shaunganley) [inline test/ui/ViewJobRolesTests.ts]: remove if not needed

File: test/ui/ViewJobRolesTests.ts [hunk 9]
```diff
@@ -0,0 +1,62 @@
+import { ViewJobRolesTestsPage } from '../ui/ViewJobRolesTestsPage';
+import { HeaderTestPage } from '../ui/HeaderTestPage';
+import { FooterTestPage } from '../ui/FooterTestPage';
+import { basepage } from './basepage';
+
+describe('View Job Roles Tests', function () {
+    this.timeout(70000);
```
Comment (shaunganley) [inline test/ui/ViewJobRolesTests.ts:5]: is this needed?

File: test/ui/ViewJobRolesTests.ts [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+    const page:string = 'https://jptw3amsi2.eu-west-1.awsapprunner.com/jobRoles';
```
Comment (shaunganley) [inline test/ui/ViewJobRolesTests.ts]: pass the URL in as an environment variable so we can run locally and against AWS

File: test/ui/ViewJobRolesTests.ts [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+    it('Job roles page should load', async function () {
+        await ViewJobRolesTestsPage.loadPage(page);
+        await ViewJobRolesTestsPage.assertJobRolesTitle();
+    });
+
+    it('Footer links should work', async function () {
+        try {
+            await ViewJobRolesTestsPage.loadPage(page);
+            await FooterTestPage.clickFacebook();
```
Comment (shaunganley) [inline test/ui/ViewJobRolesTests.ts]: these should all be ViewJobRolesTestsPage

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: test/ui/ViewJobRolesTests.ts (hunk 9 above)
Comment (ReannaMcE) [reply test/ui/ViewJobRolesTests.ts:5]: It doesn't work without this as the tests timeout too quickly

**Review by shaunganley — APPROVED** [review]

**Pull request merged**